deviceaccess.py

```diff
@@ -1229,5 +1228,5 @@ def write(self, registerPath: str, dataToWrite: np.ndarray | np.number, wordOffs
             array = dataToWrite
 
         accessModeFlags = [] if accessModeFlags is None else accessModeFlags
-        self._device.write(array, registerPath, numberOfElements,
+        self._device.write(array, registerPath, array.size,
```

Comment:
This implementation is only covering the case that the input data is smaller than the register size. The previous implementation was only covering that the data is bigger. You need both. It has to be min(numberOfElements, array.size)